src/resolvers/viewsInfo.ts

```diff
@@ -123,18 +144,25 @@ export class VideoViewsInfosResolver {
 }
 
 const mapMostViewedArray = (views: Record<string, number>, limit?: number) =>
-  Object.keys(views)
-    .map((id) => ({ id, views: views[id] }))
-    .sort((a, b) => (a.views > b.views ? -1 : 1))
-    .slice(0, limit)
+  views
+    ? Object.keys(views)
+        .map((id) => ({ id, views: views[id] }))
+        .sort((a, b) => (a.views > b.views ? -1 : 1))
+        .slice(0, limit)
+    : []
 
 const filterAllViewsByPeriod = (ctx: OrionContext, period?: number): Partial<UnsequencedVideoEvent>[] => {
-  const views = ctx.viewsAggregate.getAllViewsEvents()
+  const views = [...ctx.viewsAggregate.getAllViewsEvents()].reverse()
```

Comment:
Let's assume there will be a lot of events and avoid this unpack&reverse. We can just use an old regular `for` loop:

```ts
for (let i = events.length - 1; i >= 0; i--) {
  // ...
}
```